pkgs/build-support/fetchgit/default.nix

```diff
@@ -55,13 +55,16 @@ in
 */
 
 assert deepClone -> leaveDotGit;
-assert nonConeMode -> (sparseCheckout != "");
+assert nonConeMode -> !(sparseCheckout == "" || sparseCheckout == []);
 
 if md5 != "" then
   throw "fetchgit does not support md5 anymore, please use sha256"
 else if hash != "" && sha256 != "" then
   throw "Only one of sha256 or hash can be set"
 else
+# Added 2022-11-12
+lib.warnIf (builtins.isString sparseCheckout)
+  "Please provide files/patterns for sparseCheckout as a list of strings, passing a (multi-line) string is deprecated"
```

Comment:
It seems to depend on cone/non-cone mode. The latter only accepts directories.

> The subset of files is chosen by providing a list of directories in cone mode (which is recommended), or by providing a list of patterns in non-cone mode.

(git-sparse-checkout(1))

But if you think it is the more common case (it probably is since it is recommended), then I can change it.

Reply:
> It seems to depend on cone/non-cone mode. The latter only accepts directories.

Correct, but I think we can use directories here, since non-cone mode is deprecated, and can be removed from Git at any time. Thoughts?

Reply:
applied